src/components/eRACommons.js

```diff
@@ -0,0 +1,212 @@
+import _ from 'lodash';
+import * as qs from 'query-string';
+import React from 'react';
+import { a, button, div, hh, label, span } from 'react-hyperscript-helpers';
+import { AuthenticateNIH, Researcher } from '../libs/ajax';
+import { Config } from '../libs/config';
+import { Storage } from '../libs/storage';
+
+
+export const eRACommons = hh(class eRACommons extends React.Component {
+
+  state = {
+    isAuthorized: false,
+    expirationCount: 0,
+    nihUsername: '',
+    nihError: false,
+    isHovered: false
+  };
+
+  componentDidMount = async () => {
+    if (this.props.location !== undefined && this.props.location.search !== '') {
+      this.authenticateAsNIHFCUser(this.props.location.search);
+    } else {
+      this.getResearcherProperties();
+    }
+  };
+
+  onMouseEnter = () => {
+    this.setState({ isHovered: true });
+  };
+
+  onMouseLeave = () => {
+    this.setState({ isHovered: false });
+  };
+
+  authenticateAsNIHFCUser = async (searchArg) => {
+    const currentUserId = Storage.getCurrentUser().dacUserId;
+    let isFcUser = this.verifyUser();
+    if (!isFcUser) {
+      Researcher.getPropertiesByResearcherId(currentUserId).then(
+        (response) => isFcUser = this.registerUserToFC(response),
+        () => this.setState({ nihError: true })
+      );
+    }
+    if (isFcUser) {
+      const parsedToken = qs.parse(searchArg);
+      this.verifyToken(parsedToken).then(
+        (decodedNihAccount) => {
+          AuthenticateNIH.saveNihUsr(decodedNihAccount).then(
+            () => this.getResearcherProperties(),
+            () => this.setState({ nihError: true })
+          );
+        },
+        () => this.setState({ nihError: true })
+      );
+    }
+  };
+
+  getResearcherProperties = async () => {
+    const currentUserId = Storage.getCurrentUser().dacUserId;
+    Researcher.getPropertiesByResearcherId(currentUserId).then(response => {
+      const isAuthorized = _.isNil(response.eraAuthorized) ? false : JSON.parse(response.eraAuthorized);
```

Comment:
I think here would be more readable with `get()` from lodash. Unless I missing something here.